Guava setters notnull validations

vicayala82 opened the pull request:
## Summary
-Add notnull setters conditions with Preconditions class in Book and User class
## Trello Card
https://trello.com/c/TD55GjUH/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

Commit:
-Add notnull setters conditions with Preconditions class in Book and User class
card https://trello.com/c/TD55GjUH/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

Commit:
-Update Clases with mediaType and RestController notation
card https://trello.com/c/tFlON2s4/38-haciendo-una-request-con-postman

Commit:
Resolved merge conflicts update branch

juandbc commented on training/src/main/java/wolox/training/models/Book.java:
```diff
@@ -0,0 +1,167 @@
+    }
+
+    public void setPages(int pages) {
+        this.pages = pages;
```
Here you should check that book's pages are greater than 0 Guava has additional methods to achieve that

vicayala82 replied:
Implemented guava validation for int property.

Commit:
-Implemented guava validation for pages in book entity
-Implemented swagger annotations for book entity
https://trello.com/c/TD55GjUH/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

Commit:
-Implemented FIX guava validation for pages in book entity
-Implemented swagger annotations for book entity
https://trello.com/c/TD55GjUH/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

vicayala82 committed:
Merge branch 'main' into guava-setters-validations
